### DISCUSSION TITLE
Pingpong to laravel modules migrated issue

### DISCUSSION THREAD

**Discussion by jifinbiz** [nWidart/laravel-modules#1789] [Q&A]
I was using laravel version 5.2 with pingpong and i have 3 modules working perfectly fine. Now i upgraded laravel to 5.4 where pingpong is no more compatible. so i used laravel-module which is replacement but its giving me error. Please help.

(1/1) InvalidArgumentExceptionRoute [leave-home] not defined.
--
in UrlGenerator.php line 304
at UrlGenerator->route('leave-home', array(), true)in helpers.php line 709
at route('leave-home')in ConsoleController.php line 74

**Comment by dcblogdev** [comment]
that error is looking for a named route but not finding it, can you share the route from the routes file? 

It may be best to install a fresh copy of Laravel and go through the install for modules and then generate a new module and many copy over your classes and adjust as needed.

**Reply by jifinbiz** [reply]
Route::get('/', ['as' => 'leave-home', 'uses' => 'LeaveRequiredServicesController@index']);
 
 This is the route inside modules/leave/http/routes.php

**Comment by dcblogdev** [comment]
Try:

```php
Route::get('/', 'LeaveRequiredServicesController@index')->name('leave-home');
```

**Reply by jifinbiz** [reply]
a column bracket here
Route::get('/', 'LeaveRequiredServicesController@index'])->name('leave-home');

**Reply by jifinbiz** [reply]
Route::get('/', 'LeaveRequiredServicesController@index')->name('leave-home');
is this correct?

**Reply by dcblogdev** [reply]
yes

**Reply by jifinbiz** [reply]
still same issue.  while replacing pingpong with laravel-modules is there anything we need to do

**Reply by jifinbiz** [reply]
If we create new project in laravel 5.4  then can we import my modules to that?

**Reply by dcblogdev** [reply]
If you're upgrading I would advice to use Laravel 11 instead of 5.4

do you have a sample module I could look at to see the differences?
